src/storage/hummock_sdk/src/table_stats.rs

```diff
@@ -12,10 +12,102 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use std::borrow::Borrow;
+use std::collections::{HashMap, HashSet};
+
+use risingwave_pb::hummock::{HummockVersion, TableStats as ProstTableStats};
+
+use crate::compaction_group::hummock_version_ext::HummockVersionExt;
+
+pub type TableStatsMap = HashMap<u32, TableStats>;
+
 #[derive(Default, Debug, Clone)]
 pub struct TableStats {
-    pub total_key_size: usize,
-    pub total_value_size: usize,
-    pub total_key_count: usize,
-    pub stale_key_count: usize,
+    pub total_key_size: i64,
+    pub total_value_size: i64,
+    pub total_key_count: i64,
+    pub stale_key_count: i64,
+}
+
+impl From<&TableStats> for ProstTableStats {
+    fn from(value: &TableStats) -> Self {
+        Self {
+            total_key_size: value.total_key_size,
+            total_value_size: value.total_value_size,
+            total_key_count: value.total_key_count,
+            stale_key_count: value.stale_key_count,
+        }
+    }
+}
+
+impl From<TableStats> for ProstTableStats {
+    fn from(value: TableStats) -> Self {
+        (&value).into()
+    }
+}
+
+impl From<&ProstTableStats> for TableStats {
+    fn from(value: &ProstTableStats) -> Self {
+        Self {
+            total_key_size: value.total_key_size,
+            total_value_size: value.total_value_size,
+            total_key_count: value.total_key_count,
+            stale_key_count: value.stale_key_count,
+        }
+    }
+}
+
+impl TableStats {
+    pub fn add(&mut self, other: &TableStats) {
```

Comment:
Do we also need to decrease the stat values when SSTs are vacuumed?

Reply:
- For stale SSTs tracked in version, KV that is not dropped doesn't affect stats, because they are just moved to new SSTs. Compaction only reports stats change caused by dropped key.
- For orphan SSTs not tracked in version, they have never affected stats change.